src/status_im/contexts/shell/share/wallet/view.cljs

```diff
@@ -0,0 +1,89 @@
+(ns status-im.contexts.shell.share.wallet.view
+  (:require
+    [quo.core :as quo]
+    [react-native.core :as rn]
+    [react-native.platform :as platform]
+    [react-native.share :as share]
+    [reagent.core :as reagent]
+    [status-im.contexts.shell.share.style :as style]
+    [status-im.contexts.wallet.common.sheets.network-preferences.view :as network-preferences]
+    [status-im.contexts.wallet.common.utils :as utils]
+    [utils.i18n :as i18n]
+    [utils.image-server :as image-server]
+    [utils.re-frame :as rf]))
+
+(def qr-size 500)
+
+(defn- share-action
+  [address share-title]
+  (share/open
+   (if platform/ios?
+     {:activity-item-sources [{:placeholder-item {:type    "text"
+                                                  :content address}
+                               :item             {:default {:type "text"
+                                                            :content
+                                                            address}}
+                               :link-metadata    {:title share-title}}]}
+     {:title   share-title
+      :subject share-title
+      :message address})))
+
+(defn- open-preferences
```

Comment:
imo: it's better to pass the deref'd value unless you need to change it, that way the function just takes a param and doesn't need to know whether it's a ratom or not.

e.g
```
(defn- open-preferences
  [selected-networks]
  (rf/dispatch [:show-bottom-sheet
                {:theme :dark
                 :shell? true
                 :content
                 (fn []
                   [network-preferences/view
                    {:blur?             true
                     :selected-networks (set selected-networks)
                     :on-save           (fn [chain-ids]
                                          (rf/dispatch [:hide-bottom-sheet])
                                          (reset! selected-networks (map #(get utils/id->network %)
                                          
                                        
```
and then used as `#(open-preferences @selected-networks)`

Reply:
good shout @J-Son89, makes sense - updated 👍